pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,373 @@
+            var available_ns = get_kentities("namespaces");
+            var available_rc = get_kentities("replicationcontrollers");
+            var available_services = get_kentities("services");
+            var available_pods = get_kentities("pods");
+            var file_note = $('#deploy-app-manifest-file-note');
+            var file_note_details = $('#deploy-app-manifest-file-note-details');
+            var deploying_app_details = $('#deploy-app-deploying-details');
+            show_progress_message("Deploying App");
+
+            var btn = $('#deploy-app-start');
+            btn.prop("disabled", true);
+
+            if (isJsonString(jsonData)) {
+                var jdata = JSON.parse(jsonData);
+                if (jdata.items) {
+                    for (var i = 0; i < jdata.items.length; i++) {
+                        var ent_json = jdata.items[i];
+                        //console.log(ent_json)
+                        if (ent_json.kind === client.SERVICE) {
+                            services.push(ent_json);
+                        } else if (ent_json.kind === client.POD) {
+                            pods.push(ent_json);
+                        } else if (ent_json.kind === client.RC) {
+                            rcs.push(ent_json);
+                        } else if (ent_json.kind === client.NS) {
+                            namespaces.push(ent_json);
+                        }
+                    }
+                }
+            } else {
+                var text = "Unable to Read the file.Please check the json file. ";
+                file_note.show();
+                file_note_details.text(text);
+                return;
+            }
+
+            create_everything(namespace, services, rcs, pods)
+                .progress(function(code, response, n, total) {
+                    //console.log(" " + code + "  " + response + "  " + n + " / " + total);
+                })
+                .done(function() {
+                    /* code gets run when everything is created */
+                    var btn = $('#deploy-app-stop');
+                    btn.removeClass('btn-default');
+                    btn.addClass('btn-primary');
+                    btn.text("OK");
+                    hide_progress_message();
+                    is_deploying.parent().prepend($(Mustache.render(deploy_notification_success)));
+
+                })
+                .fail(function(ex, response) {
+                    /* ex containst the failure */
+                    var jdata = JSON.parse(response);
+                    hide_progress_message();
+                    var context = {};
+                    is_deploying.parent().prepend($(Mustache.render(deploy_notification_failure, $.extend(context, jdata))));
+
+                });
+        }
+
+        function create_everything(namespace, services, rcs, pods) {
+            var ns_json = '{"apiVersion":"v1beta3","kind":"Namespace","metadata":{"name": "' + namespace + '"}}';
```

Comment:
I would suggest defining this as an actual javascript object, and then using JSON.stringify(). That will solve any escaping and quoting issues.